### ISSUE TITLE
Kubernetes / CoreOS : ceph-mon stuck in  Waiting: CrashLoopBackOff

### ISSUE THREAD

**Issue by fjudith** [ceph/ceph-docker#681]
## Environment

```
Kubernetes: 1.6.6
CoreOS: 1437.2.0
FsType: xfs
```
Hello,

I've followed the kubernetes example after mounting xfs disks in the osd-directory (e.g. `/home/core/data/ods/`). But ceph-mon pods remain is the state `Waiting: CrashLoopBackOff`.

#### logs
```
2017-06-26T19:37:06.550463463Z ++start_mon.sh:29: get_ip(): ip -4 -o a s eth0
2017-06-26T19:37:06.550624411Z ++start_mon.sh:29: get_ip(): awk '{ sub ("/..", "", $4); print $4 }'
2017-06-26T19:37:06.551614745Z ++start_mon.sh:29: get_ip(): grep 'scope global'
2017-06-26T19:37:06.551901869Z +start_mon.sh:94: start_mon(): MON_IP=10.2.62.8
2017-06-26T19:37:06.552750828Z ++start_mon.sh:95: start_mon(): get_network eth0 4
2017-06-26T19:37:06.552760035Z ++start_mon.sh:47: get_network(): local nic=eth0
2017-06-26T19:37:06.552764151Z ++start_mon.sh:49: get_network(): local ip_version=4
2017-06-26T19:37:06.552768007Z ++start_mon.sh:51: get_network(): case "$ip_version" in
2017-06-26T19:37:06.552771988Z ++start_mon.sh:69: get_network(): is_available ip
2017-06-26T19:37:06.55277591Z ++common_functions.sh:46: is_available(): command -v ip
2017-06-26T19:37:06.552779766Z ++start_mon.sh:70: get_network(): ip -4 route show dev eth0
2017-06-26T19:37:06.55278361Z ++start_mon.sh:70: get_network(): grep proto
2017-06-26T19:37:06.553383491Z ++start_mon.sh:70: get_network(): grep -v default
2017-06-26T19:37:06.554373281Z ++start_mon.sh:70: get_network(): awk '{ print $1 }'
2017-06-26T19:37:06.554385444Z ++start_mon.sh:70: get_network(): grep /
2017-06-26T19:37:06.554575871Z +start_mon.sh:95: start_mon(): CEPH_PUBLIC_NETWORK=10.2.62.0/24
2017-06-26T19:37:06.554584469Z +start_mon.sh:96: start_mon(): ip_version=4
2017-06-26T19:37:06.554597508Z +start_mon.sh:109: start_mon(): [[ -z 10.2.62.8 ]]
2017-06-26T19:37:06.554601712Z +start_mon.sh:109: start_mon(): [[ -z 10.2.62.0/24 ]]
2017-06-26T19:37:06.554605491Z +start_mon.sh:114: start_mon(): get_mon_config 4
2017-06-26T19:37:06.554609795Z +/config.k8s.sh:11: get_mon_config(): local fsid
2017-06-26T19:37:06.556844624Z ++/config.k8s.sh:12: get_mon_config(): ceph-conf --lookup fsid -c /etc/ceph/ceph.conf
2017-06-26T19:37:06.560235229Z +/config.k8s.sh:12: get_mon_config(): fsid=c4f278fc-df9d-4501-a1ca-0a808591df87
2017-06-26T19:37:06.560245111Z +/config.k8s.sh:14: get_mon_config(): local timeout=10
2017-06-26T19:37:06.560249151Z +/config.k8s.sh:15: get_mon_config(): local monmap_add=
2017-06-26T19:37:06.560253015Z +/config.k8s.sh:17: get_mon_config(): [[ -z '' ]]
2017-06-26T19:37:06.560256732Z +/config.k8s.sh:17: get_mon_config(): [[ 10 -gt 0 ]]
2017-06-26T19:37:06.560260513Z +/config.k8s.sh:19: get_mon_config(): [[ 0 -eq 0 ]]
2017-06-26T19:37:06.560582321Z ++/config.k8s.sh:20: get_mon_config(): kubectl get pods --namespace=ceph -l daemon=mon -o template '--template={{range .items}}{{if .status.podIP}}--add {{.metadata.name}} {{.status.podIP}}:6789 {{end}} {{end}}'
2017-06-26T19:37:06.712216328Z +/config.k8s.sh:20: get_mon_config(): monmap_add='--add ceph-mon-179968503-38k1q 10.2.62.8:6789  --add ceph-mon-179968503-mxprq 10.2.108.7:6789  --add ceph-mon-179968503-p500f 10.2.105.8:6789  '
2017-06-26T19:37:06.712254052Z +/config.k8s.sh:24: get_mon_config(): ((  timeout--  ))
2017-06-26T19:37:06.712259582Z +/config.k8s.sh:25: get_mon_config(): sleep 1
2017-06-26T19:37:07.713259496Z +/config.k8s.sh:17: get_mon_config(): [[ -z --addceph-mon-179968503-38k1q10.2.62.8:6789--addceph-mon-179968503-mxprq10.2.108.7:6789--addceph-mon-179968503-p500f10.2.105.8:6789 ]]
2017-06-26T19:37:07.713298794Z +/config.k8s.sh:28: get_mon_config(): [[ -z --addceph-mon-179968503-38k1q10.2.62.8:6789--addceph-mon-179968503-mxprq10.2.108.7:6789--addceph-mon-179968503-p500f10.2.105.8:6789 ]]
2017-06-26T19:37:07.713401662Z +/config.k8s.sh:33: get_mon_config(): monmaptool --create '--add ceph-mon-179968503-38k1q 10.2.62.8:6789  --add ceph-mon-179968503-mxprq 10.2.108.7:6789  --add ceph-mon-179968503-p500f 10.2.105.8:6789  ' --fsid c4f278fc-df9d-4501-a1ca-0a808591df87 /etc/ceph/monmap-ceph
2017-06-26T19:37:07.719405878Z monmaptool: too many arguments
2017-06-26T19:37:07.71949265Z  usage: [--print] [--create [--clobber][--fsid uuid]]
2017-06-26T19:37:07.719503568Z         [--generate] [--set-initial-members]
2017-06-26T19:37:07.719508224Z         [--add name 1.2.3.4:567] [--rm name]
2017-06-26T19:37:07.719512444Z         [--feature-list [plain|parseable]]
2017-06-26T19:37:07.719516384Z         [--feature-set <value> [--optional|--persistent]]
2017-06-26T19:37:07.719520932Z         [--feature-unset <value> [--optional|--persistent]] <mapfilename>
```

**Comment by fjudith** [comment]
Sounds similar to #676 . But i'm using the very latest version

**Comment by fjudith** [comment]
The image does not reflect the fixes #676 and #678 as the `latest` leverage the branch `build-master-kraken-ubuntu-16.04` instead of `master`

**Comment by namsral** [comment]
I'm having a similar issue but with the Jewel release: `Waiting: CrashLoopBackOff`.

Environment:

```
OS: Ubuntu 16.06
Kubernetes: v1.6.5
Ceph: tag-build-master-jewel-unbutu-16.04
```

@fjudith How did you get the debug output of executed commands?

**Comment by jamstar** [comment]
same issue here (i believe).  Can we get those pushed into the latest release?

**Comment by fjudith** [comment]
@namsral : Simply add the environment variable to the ceph-mon deployment template.

```
name: DEBUG
value: verbose
```

**Comment by fjudith** [comment]
@jamstar 

i have managed to build the image on top of the `master` branch which contains #676 and #678 fixes.

However the issue remains.

As @bluecmd raised in #676, the issue seems to come from the single quotes in the following expression

### Not working
```
root@ceph-mon-0:/# monmaptool --create '--add test 1.2.3.4:1234' --fsid 1b3f7e94-fae6-4bce-ba6f-0b7c08759200 /tmp/test
```

### Working

```
monmaptool --create --add test 1.2.3.4:1234 --fsid 1b3f7e94-fae6-4bce-ba6f-0b7c08759200 /tmp/test
```

**Comment by jamstar** [comment]
so even master isn't working right now.  well hopefully this get fixed.

**Comment by fjudith** [comment]
Resolved on Kraken/Ubuntu by updating `monmaptool` command line at line 34

### reference
https://github.com/fjudith/ceph-docker/commit/f876c3ef836d2562a51e2eb90d013e3aa7c8e13c
image: fjudith/ceph-docker:latest

#### logs

```
2017-06-27T14:30:16.596805273Z ++/config.k8s.sh:20: get_mon_config(): kubectl get pods --namespace=ceph -l daemon=mon -o template '--template={{range .items}}{{if .status.podIP}}--add {{.metadata.name}} {{.status.podIP}}:6789 {{end}} {{end}}'
2017-06-27T14:30:16.74977574Z +/config.k8s.sh:20: get_mon_config(): monmap_add='--add ceph-mon-1056644374-6sg3r 10.2.105.26:6789  --add ceph-mon-1056644374-v1js0 10.2.108.26:6789  --add ceph-mon-1056644374-v4bls 10.2.62.27:6789  '
2017-06-27T14:30:16.74993384Z +/config.k8s.sh:24: get_mon_config(): ((  timeout--  ))
2017-06-27T14:30:16.750040274Z +/config.k8s.sh:25: get_mon_config(): sleep 1
2017-06-27T14:30:17.751389022Z +/config.k8s.sh:17: get_mon_config(): [[ -z --addceph-mon-1056644374-6sg3r10.2.105.26:6789--addceph-mon-1056644374-v1js010.2.108.26:6789--addceph-mon-1056644374-v4bls10.2.62.27:6789 ]]
2017-06-27T14:30:17.751559462Z +/config.k8s.sh:27: get_mon_config(): monmap_add_array=("${monmap_add}")
2017-06-27T14:30:17.751719728Z +/config.k8s.sh:29: get_mon_config(): [[ -z --addceph-mon-1056644374-6sg3r10.2.105.26:6789--addceph-mon-1056644374-v1js010.2.108.26:6789--addceph-mon-1056644374-v4bls10.2.62.27:6789 ]]
2017-06-27T14:30:17.752459357Z +++/config.k8s.sh:34: get_mon_config(): printf %s '--add ceph-mon-1056644374-6sg3r 10.2.105.26:6789  --add ceph-mon-1056644374-v1js0 10.2.108.26:6789  --add ceph-mon-1056644374-v4bls 10.2.62.27:6789  '
2017-06-27T14:30:17.752701789Z ++/config.k8s.sh:34: get_mon_config(): echo --add ceph-mon-1056644374-6sg3r 10.2.105.26:6789 --add ceph-mon-1056644374-v1js0 10.2.108.26:6789 --add ceph-mon-1056644374-v4bls 10.2.62.27:6789
2017-06-27T14:30:17.752953553Z +/config.k8s.sh:34: get_mon_config(): monmaptool --create --add ceph-mon-1056644374-6sg3r 10.2.105.26:6789 --add ceph-mon-1056644374-v1js0 10.2.108.26:6789 --add ceph-mon-1056644374-v4bls 10.2.62.27:6789 --fsid 1a33e7a4-60df-4459-983b-07d72c00ec70 /etc/ceph/monmap-ceph
2017-06-27T14:30:17.830677899Z monmaptool: monmap file /etc/ceph/monmap-ceph
2017-06-27T14:30:17.830716187Z monmaptool: set fsid to 1a33e7a4-60df-4459-983b-07d72c00ec70
2017-06-27T14:30:17.830721451Z monmaptool: writing epoch 0 to /etc/ceph/monmap-ceph (3 monitors)
```

**Comment by leseb** [comment]
@fjudith thanks, looking at your commit, I think the `echo` is not necessary.
Let me try to reproduce this. Will have a fix today.

**Comment by leseb** [comment]
@fjudith I think I have a fix, please see #688

**Comment by fjudith** [comment]
@leseb All good. Thanks
